query/excluder.go

```diff
@@ -0,0 +1,94 @@
+package query
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+//Excluder allows us to support different methods of excluding in the future
+type Excluder interface {
+	ShouldExclude(path string) bool
+}
+
+//RegexpExclude uses regular expressions to tell if a file/path should be excluded
+type RegexpExclude struct {
+	recursive  bool
+	exclusions []string
+	regex      string
+}
+
+//ShouldExclude will return a boolean denoting whether or not the path should be excluded based on the given slice of exclusions
+func (r *RegexpExclude) ShouldExclude(path string) bool {
+	var b bool
+	var ok error
+	if r.regex == "" {
+		r.buildRegex()
+	}
+
+	if b, ok = regexp.MatchString(r.regex, path); not(ok) {
+		fmt.Printf("regexp failed: %s \n" + ok.Error())
+		return false
+	}
+	return b
+}
+
+func (r *RegexpExclude) buildRegex() {
+	var regex string
+	var prev string
+	var curr string
+	for _, exclusion := range r.exclusions {
+		prev = curr
+		if strings.HasSuffix(exclusion, "/") {
```

Comment:
I like what's happening here, but I think there's a simpler way to do it. So instead of `^.git$|^.git/.*$`,  we can use `^.git(/.*)?$`.

You can simplify this loop by first [right-trimming](https://golang.org/pkg/strings/#TrimRight) any `/` and [replacing](https://golang.org/pkg/strings/#Replace) all `.` with `\\.`, then wrapping the result with `^` and `(/.*)?$`.

I suggest creating a slice where each exclusion is put into that form, and then joining that slice with [`strings.Join`](https://golang.org/pkg/strings/#Join) after the loop is done.